src/components/PostDetails.tsx

```diff
@@ -1,117 +1,147 @@
+  postComments: Comment[],
+  isLoadingComments: boolean,
+  setComments: (value: Comment[])=> void,
+  setPostComments: (value: Comment[])=> void,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post, comments, setComments,
+  isLoadingComments,
+  postComments, setPostComments,
+}) => {
+  const [isSeen, setIsSeen] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const deleteComment = async (comment: Comment) => {
+    const response = client.delete(`/comments/${comment.id}`);
+    const result = await response;
+
+    if (result === 1) {
+      setComments(comments.filter((one: Comment) => one.id !== comment.id));
+      setPostComments(postComments.filter(
+        (one: Comment) => one.id !== comment.id,
+      ));
+
+      return;
+    }
+
+    setIsError(true);
+  };
+
+  useEffect(() => {
+    setIsSeen(false);
+  }, [post]);
+
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            #
+            {post.id}
+            :
+            {' '}
+            {post.title}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {post.body}
           </p>
         </div>
 
-        <div className="block">
-          <Loader />
+        {isLoadingComments && <Loader />}
+        {!isLoadingComments
+        && (
+          <div className="block">
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+            {isError && (
+              <div className="notification is-danger" data-cy="CommentsError">
+                Something went wrong
+              </div>
+            )}
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+            {postComments.length > 0
+               && (
+                 <>
+                   <p className="title is-4">Comments:</p>
+                   {postComments.map((comment: Comment) => {
+                     return (
+                       <article
+                         key={comment.id}
+                         className="message is-small"
+                         data-cy="Comment"
+                       >
+                         <div className="message-header">
+                           <a
+                             href={`mailto:${comment.email}`}
+                             data-cy="CommentAuthor"
+                           >
+                             {comment.name}
+                           </a>
+                           <button
+                             data-cy="CommentDelete"
+                             type="button"
```

Comment:
Would be good to move this article to a separate component and destructure the comment inside it